hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -146,8 +159,14 @@ object of DatanodeDetails(with Topology Information) while trying to get the
   random node from NetworkTopology should fix this. Check HDDS-7015
  */
     return chooseDatanodesInternal(
+            Objects.isNull(usedNodes) ? Collections.emptyList()
```

Comment:
Might be a good idea to move the logic for handling the empty collection and mapping it into a private method, as I think its duplicate here for used and excluded nodes?

Reply:
@sodonnel Is it not better do the Null Check handling in one place.

Reply:
The modified code looks like:

```
 return chooseDatanodesInternal(
          Objects.isNull(usedNodes) ? Collections.emptyList()
                    : usedNodes.stream().map(node -> {
                      DatanodeDetails datanodeDetails =
                              nodeManager.getNodeByUuid(node.getUuidString());
                      return datanodeDetails != null ? datanodeDetails : node;
                    }).collect(Collectors.toList()),
            Objects.isNull(excludedNodes)
                    ? excludedNodes : excludedNodes.stream()
                    ? Collections.emptyList() : excludedNodes.stream()
                    .map(node -> {
                      DatanodeDetails datanodeDetails =
                              nodeManager.getNodeByUuid(node.getUuidString());
                      return datanodeDetails != null ? datanodeDetails : node;
                    }).collect(Collectors.toList()));
```

Could we not change this to look like:

```
private List<DatanodeDetails> validateDatanodes(List<DatanodeDetails> dns) {
  return Objects.isNull(dns) ? Collections.emptyList()
                    : dns.stream().map(node -> {
                      DatanodeDetails datanodeDetails =
                              nodeManager.getNodeByUuid(node.getUuidString());
                      return datanodeDetails != null ? datanodeDetails : node;
                    }).collect(Collectors.toList())
}

 return chooseDatanodesInternal(
  validateDatanodes(usedNodes), validateDatanode(excludedNodes);
_
```

The logic used to map both used and excluded looks to be the same to me, and its complicated enough it should be extracted into a method I think.